app.py

```diff
@@ -22,29 +36,27 @@ def bot(history):
 
         yield history
 
-with gr.Blocks() as demo:
-    with gr.Row():
-        with gr.Column(scale=1):
-            gr.Label("Griptape PDF Chat", show_label=False)
-            gr.Text(
-                "Before using chat, do the following:\n\n1. Make the OpenAI API token available via an environment "
-                "variable OPENAI_API_KEY\n2. Upload a PDF file that you'd like to chat with",
-                show_label=False
-            )
-            file_output = gr.File()
-            upload_btn = gr.UploadButton(
-                variant="primary",
-                file_types=[".pdf"],
-                file_count="single"
-            )
-        with gr.Column(scale=3):
-            chatbot = gr.Chatbot(show_label=False)
-            msg_textbox = gr.Textbox(placeholder="Send a message", show_label=False)
-
-            upload_btn.upload(chat.upload_pdf, upload_btn, file_output)
-
-            msg_textbox.submit(user, [msg_textbox, chatbot], [msg_textbox, chatbot], queue=False).then(
-                bot, chatbot, chatbot
-            )
-    demo.queue()
+# Checks if environment variables are set for the Griptape Cloud or Azure
+if "GT_STRUCTURE_ID" in os.environ and os.environ["GT_STRUCTURE_ID"]:
+    # Launch the chat interface WITH session state (Managed Environment)
+    chat = Chat_Cloud()
+    demo = gr.ChatInterface(fn=chat.send_message, additional_inputs=[gr.State(value=get_state())])
+    demo.launch(share=True)
+elif "AZURE_CLIENT_ID" in os.environ and os.environ["AZURE_CLIENT_ID"]:
+    # Launch the chat interface locally WITH Vector Store (Local Agent)
+    chat = Chat()
```

Comment:
The naming of `Chat_Cloud`, `Chat`, and `Chat_Local` is confusing. "Local" in particular is ambiguous. Maybe spell out the environments/situations more clearly in a comment above.